src/tests/example.spec.js

```diff
@@ -1,27 +1,193 @@
-        await loginPage.performLogin('standard_user', 'secret_sauce');
-        await inventoryPage.addItemToCartById(0);
-        expect(await inventoryPage.getNumberOfItemsInCart()).toBe('1');
+  const filterArr = ["az", "za", "lohi", "hilo"];
+ 
 
-        await inventoryPage.shopingCart.click();
-        expect(await shopingCartPage.cartItems.count()).toBeGreaterThanOrEqual(1);
 
-        await shopingCartPage.removeCartItemById(0);
-        await expect(shopingCartPage.cartItems).not.toBeAttached();
+  for (let element of filterArr) {
+    test(`Perform and verify sorting on the Inventory page by ${element}`, async ({
+      inventoryPage,
+    }) => {
+      //sorting names
+      const elementsNames =  await inventoryPage.productElementName.allTextContents();
+      const expectedAZ = await elementsNames.sort();
+     
+
+      //sorting prices
+      const elementsPrice = await inventoryPage.productPrice.allTextContents();
+      const elementsNumberOfPrice = await inventoryPage.selectPriceNumber(elementsPrice);
+      const expectedLoHi = await elementsNumberOfPrice.sort(function (a, b) { return a - b; });
+   
+
+      await inventoryPage.sortContainer.selectOption(element);
+      const catualElementsName = await inventoryPage.productElementName.allTextContents();
+      const actualElementsPrice = await inventoryPage.productPrice.allTextContents();
+      const actualElementsNumberOfPrice = await inventoryPage.selectPriceNumber(actualElementsPrice);
+
+      if (element === "az") {
+        expect(catualElementsName ).toEqual(expectedAZ);
+      } else if (element === "za") {
+        expect(catualElementsName).toEqual(expectedAZ.reverse());
+      } else if (element === "lohi") {
+         expect(actualElementsNumberOfPrice).toEqual(expectedLoHi );
+      } else if (element === "hilo") { 
+        expect(actualElementsNumberOfPrice).toEqual( expectedLoHi.reverse() );
+      }
+    
     });
+  }
+
+  test("Add several random products to the Shopping Cart and verify products are displayed correctly", async ({
+    inventoryPage,
+    shopingCartPage,
+  }) => {
+    //select all "Add to cards btns"
+    const addToCartBtns = await inventoryPage.inventoryItems.allTextContents();
+
+    // select random product index
+    const randomProductNumbers =  Math.floor(Math.random()*addToCartBtns.length) +1 
+
+    // creation an array of random number of products
+
+    let arrayOfProductRandomIndex = [];
+
+    for( let i = 0; i < randomProductNumbers; i++){
+        const randomFunction = async function(){
+            let randomProductIndex = Math.floor(Math.random()*addToCartBtns.length)+1;
+            // const arrayItemExisting = randomArrayOfProductItems.includes(randomProductIndex)
+            if(arrayOfProductRandomIndex.includes(randomProductIndex)){
+                randomFunction()
+            }else {
+                arrayOfProductRandomIndex.push(randomProductIndex)
+            }
+        }
+        randomFunction()
```

Comment:
I would suggest to decide if we want to add to card each index instead of  doing this ^^. I am not sure, what is going on here :) So suggestion: `for (const productIndex of addToCartBtns) if (random(true, false)) addtoCard(productIndex)` <- here we need to create such kind of random(true, false). This should be easier to understand.